tests/overlay/setup-overlay

```diff
@@ -0,0 +1,51 @@
+#/bin/bash
+#set -x
+
+BASEDIR=$1
+CONTEXT=$2
+
+setup () {
+    # Create and populate the lower directory.
+    # This would be e.g. a Fedora image in the container scenario.
+    mkdir -p $BASEDIR/lower/readdir $BASEDIR/lower/writedir $BASEDIR/lower/noaccessdir
+    echo read > $BASEDIR/lower/readfile
+    echo write > $BASEDIR/lower/writefile
+    echo noaccess > $BASEDIR/lower/noaccessfile
+
+    # Label the lower directory with a type readable but not writable by
+    # containers.  I just picked a type from base_ro_file_type.
+    chcon -R -t test_overlay_files_ro_t $BASEDIR/lower
+    chcon -t test_overlay_files_noaccess_t $BASEDIR/lower/noaccessfile $BASEDIR/lower/noaccessdir
+    # Writefile on lower is writable so that nocontext mount can also trigger
+    # copy up and write.
+    chcon -t test_overlay_files_rwx_t $BASEDIR/lower/writefile $BASEDIR/lower/writedir
+
+    # Create a directory per container.
+    mkdir -p $BASEDIR/container1
+
+    # Create upper, work, and overlay directories per container.
+    mkdir -p $BASEDIR/container1/upper $BASEDIR/container1/work $BASEDIR/container1/merged
+    chcon -R unconfined_u:object_r:test_overlay_files_ro_t:s0:c10,c20 $BASEDIR/container1
+
+    # Label the container directories to match the container context.
+    # This is simply to ensure correct label inheritance on new file
+    # creation and to prevent permission problems when overlayfs
+    # internally creates files on behalf of the current process.
+    chcon -R unconfined_u:object_r:test_overlay_files_rwx_t:s0:c10,c20 $BASEDIR/container1/upper $BASEDIR/container1/work $BASEDIR/container1/merged
```

Comment:
@stephensmalley Was chatting with dan walsh about it. We were thinking that it should make sense that newly create file gets a label as if task was writing to upper/ dir for non-context mount case.
 
I am wondering can we determine the label of newly created file in advance and use that label to do create_sid. Dan mentioned that we probably need 4 things. "task, parent dir, object type and file name" to determine the label of the newly created object. 

So in case of non-context mount we will determine label based on above 4 and in the context mount lablel comes from context= as usual and set it in create_sid.

I am wondering if SELinux already has a function to determine the label based on above 4 parameters. I should probably read code.